flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -292,18 +303,62 @@ public void alterDatabase(String databaseName, CatalogDatabase newDatabase, bool
 		checkNotNull(newDatabase, "newDatabase cannot be null");
 
 		// client.alterDatabase doesn't throw any exception if there is no existing database
-		if (!databaseExists(databaseName)) {
+		Database hiveDB;
+		try {
+			hiveDB = getHiveDatabase(databaseName);
+		} catch (DatabaseNotExistException e) {
 			if (!ignoreIfNotExists) {
 				throw new DatabaseNotExistException(getName(), databaseName);
 			}
 
 			return;
 		}
-
-		Database newHiveDatabase = instantiateHiveDatabase(databaseName, newDatabase);
+		Map<String, String> params = hiveDB.getParameters();
```

Comment:
I can see `HiveCatalog` becomes big and big, can we split database related codes to a separate class?